sdk/storage/azure-storage-blob/samples/copy_blob.py

```diff
@@ -0,0 +1,56 @@
+# coding: utf-8
+
+# -------------------------------------------------------------------------
+# Copyright (c) Microsoft Corporation. All rights reserved.
+# Licensed under the MIT License. See License.txt in the project root for
+# license information.
+# --------------------------------------------------------------------------
+
+"""
+FILE: copy_blob.py
+DESCRIPTION:
+    This sample demos how to copy a blob from a URL.
+USAGE: python copy_blob.py
+"""
+
+from __future__ import print_function
+import os
+import sys
+import time
+from azure.storage.blob import BlobServiceClient
+
+def main():
+    try:
+        CONNECTION_STRING = os.environ['AZURE_STORAGE_CONNECTION_STRING']
+
+    except KeyError:
+        print("AZURE_STORAGE_CONNECTION_STRING must be set.")
+        sys.exit(1)
+
+    is_finished = False
+    blob_service_client = BlobServiceClient.from_connection_string(CONNECTION_STRING)
+    source_blob = "http://www.gutenberg.org/files/59466/59466-0.txt"
+    copied_blob = blob_service_client.get_blob_client("mycontainer", '59466-0.txt')
+    # Copy started
+    copied_blob.start_copy_from_url(source_blob)
+    for i in range(10):
+        props = copied_blob.get_blob_properties()
+        status = props.copy.status
+        print("Copy status: " + status)
+        if status == "success":
+            # Copy finished
+            is_finished = True
+            break
+        time.sleep(10)
+
+    if not(is_finished):
```

Comment:
if you set `status = None` before the loop, this could check `if status != 'success':` and get rid of `is_finished` altogether. 

Either way, it seems like the operation could complete after 90 seconds is passed, and an abort will still be tried?

Reply:
Yes. the copy operation may need more than 100s. I added this in order to demo how to abort the copy operation.

Reply:
@xiangyan99 My point is that the sleep happens at the *end* of a loop iteration, and so there is no check after 90s. If the job finished at 95s, it will be considered "not finished" even though it finished in under 100 seconds.